### PR TITLE
can: add tja1042 transceiver driver

### PR DESCRIPTION
**Pull request by vincent-d**
Follow up, based on #5793 

This is the TJA1042 (http://www.nxp.com/documents/data_sheet/TJA1042.pdf) HS CAN transceiver driver.

### CODE REVIEW

**Comment by vincent-d** [comment]
Rebased and updated, added an extra test app. Still need to test it.

File: drivers/include/tja1042.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2016  OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    tja1042 TJA1042
```
Comment (smlng) [inline drivers/include/tja1042.h]: should be `drivers_tja1042`

File: drivers/include/tja1042.h [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * @ingroup     trx_can
```
Comment (smlng) [inline drivers/include/tja1042.h]: multiple groups are possible, add `@ingroup drivers` here, too

File: drivers/include/tja1042.h [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+#include "periph/gpio.h"
+#include "can/can_trx.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * tja1042 can trx descriptor
```
Comment (smlng) [inline drivers/include/tja1042.h]: missing `@brief`

File: tests/can_trx/Makefile [hunk 4]
```diff
@@ -0,0 +1,20 @@
+export APPLICATION = can_trx
+include ../Makefile.tests_common
+
+
+CFLAGS += -DDEVELHELP
```
Comment (smlng) [inline tests/can_trx/Makefile]: can be removed, set in tests_common

**Comment by vincent-d** [comment]
@smlng comments addressed

File: drivers/tja1042/tja1042.c [hunk 5]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016  OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tja1042 trx
```
Comment (smlng) [inline drivers/tja1042/tja1042.c]: should be `@ingroup drivers_tja1042` only

Comment (smlng) [reply drivers/tja1042/tja1042.c]: though also these files are not parsed by doxygen, anyway.

Comment (vincent-d) [reply drivers/tja1042/tja1042.c]: addressed

**Comment by smlng** [comment]
looks good, please squash.

**Comment by vincent-d** [comment]
All green, go!

**Pull request merged**